lib/instructor/openai/client.rb

```diff
@@ -0,0 +1,44 @@
+OpenAI.configure do |config|
+  config.access_token = ENV.fetch('OPENAI_API_KEY')
+end
+
+module Instructor
+  module OpenAI
+    class Client
+      def initialize
+        @client = ::OpenAI::Client.new do |f|
+          f.response :logger, Logger.new($stdout), bodies: true if ENV['OPENAI_LOG'] == 'debug'
+        end
+      end
+
+      def chat(parameters:, response_model:)
+        func = generate_function(response_model)
+        params = parameters.merge(tools: [func])
```

Comment:
one of the features is the need to have: https://jxnl.github.io/instructor/concepts/patching/